### DISCUSSION TITLE
Question on xdrip+ upload to nightscout

### DISCUSSION THREAD

**Discussion by Npavkov** [NightscoutFoundation/xDrip#2814] [Q&A]
How do I get xdrip+ to upload CALIBRATED data to nightscout on an nightscout address like this : https://aaaaaaaaa.ns.10be.de:99999/ ? I can not seem to get this to work in xdrip+ 'upload to cloud'???

**Comment by Navid200** [comment]
Do you mean that you have two different sets of data, calibrated and uncalibrated?

What is your hardware data source?

**Comment by Npavkov** [comment]
I am using diabox as input to xdrip+ thru nighscout folllower setting in xdrip+. But right now I am having diabox send data to xdrip+ and nightscout directly. I would like to have diabox continue to send calibrated data to xdrip+ and then have xdrip+ RECALIBRATE the data and send it to nightscout at the address I specified in my orig post. that https address has a password to authenticate to the nightscout https address.

**Reply by Navid200** [reply]
And it doesn't work when you try it?

**Reply by Navid200** [reply]
Any Nightscout server has instructions for setting up an uploader.
How does the 10be site tell you to set up xDrip to upload to it?

**Comment by Npavkov** [comment]
No it doesnot seem to work using the address thru cloud upload syntax, I do not see any instructions on nightscout page to tell how to get xdrip+ to upload data to it. in diabox I enter the https address and a password to perform the upload to nightscout.

**Reply by Navid200** [reply]
10be is a paid service.  Are you a customer?
They must tell you how to set up your uploader no matter what it is.

**Comment by Npavkov** [comment]
yes I am a customer of the service and 10be says that to upload url should be the nightscout page name and a assigned token for each subject. which would be the https address with the token but that doesnot work in xdrip+ cloud upload entry when I put it in, maybe I could be entering it in cloud upload wrong way or syntax, but not sure.
I really would like to get diabox to send the raw data to xdrip+ app and then have xdrip+ calibrate the readings to get it to shhow my actually BG levels in nightscout.

**Reply by Navid200** [reply]
To upload from xDrip to any Nightscout, all you need is the hostname and the API_SECRET.
This is how you use those two to set up xDrip as a Nightscout uploader:
https://navid200.github.io/xDrip/docs/Nightscout-Settings.html

**Comment by Npavkov** [comment]
Thank you so much for your help, that syntax for cloud upload worked perfectly.... I have been struggling with the settings for that for days. 
One more question, now how do I get xdrip+ to calibrate the readings it receives from diabox to improve the reading accuracy?

**Reply by Navid200** [reply]
I am not in favor of that capability.
xDrip should be user friendly for all including those who are not very tech savvy.  Manipulating calibration is not what we should easily allow in xDrip.  It should only be done by someone who knows what they are doing.  that is not 90% of our user community.  I'm affraid that's not even 50% of our user population.

So, I leave this to be answered by others.

**Comment by Npavkov** [comment]
the calibration being done in diabox does not seem to be very accurate for very long. I was hoping xdrip+ would be better at the calibration to allow for more accurate BG readings being sent to nightscout.

**Reply by Navid200** [reply]
Your questions worry me.
xDrip should not be used for dosing decisions.

Having calibrations applied at two completely different points of a control system is a very bad design choice.
Calibration schemes are almost always created with the assumption that they are the only calibration mechanism present in the system.

Whose idea is this?  Is it published somewhere?  Have you discussed it with your doctor?

**Reply by Npavkov** [reply]
No I did not find it published anywhere, and no there has been no discussion with my doctor. It is just an effort for me to obtain more accurate readings.

**Comment by Npavkov** [comment]
Also, is there a way to control how often xdrip+ sends data to nightscout?
